### PR TITLE
Move more things out of FrigateApp

### PR DESCRIPTION
**Pull request by gtsiam**
This should (hopefully) be the last tangent to rewriting `FrigateApp` to properly manage all its subprocesses. I moved all the `FrigateConfig` parsing out of `FrigateApp` so that it ~~only~~ mainly deals with all the subprocesses it spawns.

**Changes:**

- Moved functionality of `FrigateApp.init_config()` into `FrigateConfig.load()`
- `main()` is now responsible for loading the config (and potentially verifying it, if cli says to do so).
- Moved `PlusApi` to a private attr in `FrigateConfig`.
- Moved `SIGTERM` registration to `main()`.
- Set the log thread up earlier, since `FrigateConfig.load()` is now called in main.

---

**Changes (2):**
- Removed logging from signal handlers, since python's logging module functions tend to not be [re-entrant](https://docs.python.org/3/library/logging.html#thread-safety).
- Added code to reopen stdout & stderr on fork. This helps prevent a deadlock I was encountering.

### CODE REVIEW

File: frigate/app.py
Comment (hawkeye217) [inline frigate/app.py:737]: `SIGINT` works in the devcontainer (ctrl-c), but Docker sends `SIGTERM`. Is this no longer necessary?

Comment (gtsiam) [reply frigate/app.py:737]: I'm not sure just yet to be honest. I'd like to unravel the multiprocessing mess first.

Currently frigate never really shuts down cleanly (it always hits the 10 second timeout in docker). So until I can get the devcontainer to shut down in under a second, it's hard to test. I _think_ it's unnecessary, but I'd rather come back to it later.

PS: What I mean by "mess" is to try running with `mp.set_start_method("spawn", force=True)`. (Here's the reason [why](https://github.com/python/cpython/issues/84559)). We're hitting all kinds of undefined behaviors right now.

Comment (NickM-27) [reply frigate/app.py:737]: with a docker timeout of 30 seconds Frigate shuts down cleanly before the timeout for me so I think it would be good to ensure that the behavior stays the same after this PR is merged. There is another user contributing a separate change that will likely remove it anyway

Comment (gtsiam) [reply frigate/app.py:737]: Huh. It does stop, but it takes ~120 seconds for me (vs ~30 in 0.14). This needs some investigation then.

Comment (gtsiam) [reply frigate/app.py:737]: Okay. As far as the signal handler goes, I say we leave it here. Since there's another change on the way that might remove it anyway, might as well leave it for then.

But I'll leave the final call to you.

**Comment by gtsiam** [comment]
As far as the slow closing goes, I can't seem to reproduce it now. My network is incredibly unreliable (telco cartels and such), so that might be what I was seeing.

There were two deadlocks though, which should now be fixed ([f407776](https://github.com/blakeblackshear/frigate/pull/13897/commits/f407776e9a37425dacbde8ee6f83562ef600dd5f) and [0a11ed3](https://github.com/blakeblackshear/frigate/pull/13897/commits/0a11ed378184434647e1edd9ad5a8002f3a17e58)).

File: frigate/app.py
Comment (NickM-27) [reply frigate/app.py:737]: agreed, leave it in for now

**Review by NickM-27 — APPROVED** [review]

**Review by hawkeye217 — APPROVED** [review]

File: frigate/app.py
Comment (gtsiam) [reply frigate/app.py:737]: It's in __main__.py

Comment (gtsiam) [reply frigate/app.py:737]: Actually we probably need to habdle SystemExit in addition to keyboardinterrupt now. Give me a bit to fix that.

Comment (gtsiam) [reply frigate/app.py:737]: Done.

**Pull request merged**
